Userland/Utilities/sed.cpp

```diff
@@ -839,6 +877,29 @@ static ErrorOr<CycleDecision> apply(Command const& command, StringBuilder& patte
         TRY(write_pattern_space(input, pattern_space));
         break;
     }
+    case 'y': {
+        // FIXME: Escaping in the transform strings doesn't work properly
+        auto pattern_space_sv = pattern_space.string_view();
+        auto const& y_args = command.arguments->get<YArguments>();
+        VERIFY(y_args.characters.length() == y_args.replacements.length());
+
+        HashMap<char, char> replacement;
+        for (size_t i = 0; i < y_args.characters.length(); i++) {
+            TRY(replacement.try_set(y_args.characters[i], y_args.replacements[i]));
+        }
+
+        StringBuilder new_string;
+        for (size_t i = 0; i < pattern_space.length(); i++) {
+            if (replacement.contains(pattern_space_sv[i]))
+                new_string.append(replacement.get(pattern_space_sv[i]).value());
+            else
+                new_string.append(pattern_space_sv[i]);
+        }
+
+        pattern_space.clear();
+        pattern_space.append(new_string.to_byte_string());
+        break;
+    }
```

Comment:
Since you added tests for the `q` change: Could we have tests for this feature too? :)